packages/runtime-dom/src/fallthroughAttrs.ts

```diff
@@ -0,0 +1,13 @@
+import { isOn } from '@vue/shared'
+
+type Data = Record<string, any>
+
+export const getFallthroughAttrs = (attrs: Data) => {
+  let res: Data | undefined
+  for (const key in attrs) {
+    if (key === 'class' || key === 'style' || isOn(key)) {
+      ;(res || (res = {}))[key] = attrs[key]
```

Comment:
maybe you can avoid in each iteration ask if the variable "res" has value or not to initialize it

Reply:
This is not my code: I simply copied it.

There must be a reason for this construction. I think that the ‘hot path’ in this function is to skip all attrs and return undefined. It might hurt performance to create a new object unnecessarily.